### ISSUE TITLE
BleCannotSetCharacteristicNotificationException (Caused by: BleGattDescriptorException status=3 (0x03))

### ISSUE THREAD

**Issue by fradali** [Polidea/RxAndroidBle#291]
Hello,
I'm trying to connect in DFU mode my device, I've checked some old issues, but it didn't help me at all.
The device I'm using has already a notify characteristic with a Client Characteristic Config descriptor.
So what I'm trying to do is:

- Connect to device
- Execute a command to switch to DFU mode
- Disconnect the device to search for the device(in DFU mode) with a new Mac Address 
- Start the update of Firmware using the device connected 

The exception is happening in this part of code:

`

        if (isInsoleConnected()) {
            Subscription subscription = notifyCharacteristic(mZtTrackDevice.getDevice(), mZtTrackDevice.getDfuControlPointCharacteristic().getUuid())
                    .doOnNext(observable -> {
                        if (callback != null) {
                            callback.call();
                        }
                    })
                    .onErrorResumeNext(throwable -> {
                        Log.e(TAG, "notifyDfuControlPointCharacteristic: ", throwable); // NON-NLS
                        return Observable.empty();
                    })
                    .subscribe();
            mCharacteristicSubscription.add(subscription);
        }
    
`

> 

The Log is attached below
[issue.txt](https://github.com/Polidea/RxAndroidBle/files/1362403/issue.txt)

**Comment by dariuszseweryn** [comment]
As you can see in the logs:
`Caused by: BleGattDescriptorException{macAddress=F5:56:71:04:1B:60, status=3 (0x03 -> https://android.googlesource.com/platform/external/bluetooth/bluedroid/+/android-5.1.0_r1/stack/include/gatt_api.h), bleGattOperationType=BleGattOperation{description='DESCRIPTOR_WRITE'}}`

It means that the `BluetoothGattCallback.onDescriptorWrite()` was called with `status=3`. Going to https://android.googlesource.com/platform/external/bluetooth/bluedroid/+/android-5.1.0_r1/stack/include/gatt_api.h and searching for `0x03` gives three results:
* `#define  GATT_WRITE_NOT_PERMIT               0x03`
* `#define  GATT_RSP_MTU                        0x03`
* `GATT_FORMAT_NIBBLE,         /* 0x03 nibble */`
Of which only the first has sense as writing a descriptor is not an `MTU` related operation nor it is related to format in any way.
TL;DR—the peripheral you are trying to update does not a permit to write the `Client Characteristic Config Descriptor` or it is some kind of Android BLE stack bug. Either way it does not seem to be possible to fix it from the library side.

Best Regards
